packages/sourcecred/src/core/ledger/policies/balanced.js

```diff
@@ -54,42 +55,104 @@ export type BalancedPolicy = {|
  * scores.
  */
 export function balancedReceipts(
-  budget: G.Grain,
-  identities: ProcessedIdentities
+  policy: BalancedPolicy,
+  credGrainView: CredGrainView,
+  effectiveTimestamp: TimestampMs
 ): $ReadOnlyArray<GrainReceipt> {
-  const totalCred = sum(identities.map((x) => x.lifetimeCred));
-  const totalEverPaid = G.sum(identities.map((i) => i.paid));
+  if (policy.numIntervalsLookback < 0) {
+    throw new Error(
+      `numIntervalsLookback must be at least 0, got ${policy.numIntervalsLookback}`
+    );
+  }
+  if (!Number.isInteger(policy.numIntervalsLookback)) {
+    throw new Error(
+      `numIntervalsLookback must be an integer, got ${policy.numIntervalsLookback}`
+    );
+  }
+
+  const intervalsBeforeEffective = credGrainView
+    .intervals()
+    .filter((interval) => interval.endTimeMs <= effectiveTimestamp);
+
+  const numIntervalsLookback = ((
+    policy: BalancedPolicy,
+    intervalsLength: number
+  ) => {
+    if (
+      !policy.numIntervalsLookback ||
+      policy.numIntervalsLookback > intervalsLength
+    ) {
+      return intervalsLength;
+    } else {
+      return policy.numIntervalsLookback;
+    }
+  })(policy, intervalsBeforeEffective.length);
```

Comment:
Good opportunity for a ternary operator.
```suggestion
  const numIntervalsLookback = !policy.numIntervalsLookback ||
      policy.numIntervalsLookback > intervalsBeforeEffective.length ?
      intervalsBeforeEffective.length :
      policy.numIntervalsLookback;
```